### DISCUSSION TITLE
No example of Pioneer3 position estimation in Matlab

### DISCUSSION THREAD

**Discussion by jafphd** [cyberbotics/webots#5095] [Q&A]
Hi All, I am looking for an example in Pioneer3 (Webots R2022a) "using Matlab" to get the robot position. I tried to do it based on the weel's encoders (or GPS but the same missing examples). After days of searching, I could not find any code for that position estimation. 
I would appreciate it very much if someone could let me know where to faun an example please! Thanks!

-------------------------------
I tried defining:
    left_wheel   = wb_robot_get_device('left wheel');
    right_wheel = wb_robot_get_device('right wheel');
    position_sensor_left   = wb_motor_get_position_sensor(left_wheel);  
    position_sensor_right = wb_motor_get_position_sensor(right_wheel);
and reading:
    position_sensor_left.getValue()
    position_sensor_right.getValue()

but I got errors like Struct contents reference from a non-struct array object.

**Comment by stefaniapedrazzi** [comment]
The correct MATLAB function to read the `PositionSensor value is:
```
left_value = wb_position_sensor_get_value(position_sensor_left)
right_value = wb_position_sensor_get_value(position_sensor_right)
```

Here you can find the MATLAB documentation:
https://www.cyberbotics.com/doc/reference/positionsensor?tab-language=matlab#wb_position_sensor_get_value

Here you can find a sample MATLAB controller for the Robotis OP2 robot that can help you understand how to use the MATLAB API:
https://github.com/cyberbotics/webots/blob/master/projects/robots/robotis/darwin-op/controllers/symmetry_matlab/symmetry_matlab.m

**Reply by jafphd** [reply]
Thank you for your reply @stefaniapedrazzi . Best!
